cpp/include/cudf/join.hpp

```diff
@@ -24,3 +24,4 @@
 #include <cudf/join/hash_join.hpp>
 #include <cudf/join/join.hpp>
 #include <cudf/join/mixed_join.hpp>
+#include <cudf/join/sort_merge_join.hpp>
```

Comment:
Let's avoid adding new includes to a deprecated header. New users of this API should use the new headers.
```suggestion
```